src/unix/thread.c

```diff
@@ -461,9 +461,12 @@ typedef struct uv_semaphore_s {
   unsigned int value;
 } uv_semaphore_t;
 
+#ifdef __GLIBC__
```

Comment:
Looking at the changes that were made above, I think we also need to check for `__MVS__` now, where we always seem to use this pattern

Reply:
I'd guard it on `#if platform_needs_custom_semaphore`.  Can you drop the empty lines around the STATIC_ASSERT?